pkg/workload/schemachange/optype.go

```diff
@@ -0,0 +1,209 @@
+	dropConstraint:          (*operationGenerator).dropConstraint,
+	dropIndex:               (*operationGenerator).dropIndex,
+	dropSequence:            (*operationGenerator).dropSequence,
+	dropTable:               (*operationGenerator).dropTable,
+	dropView:                (*operationGenerator).dropView,
+	dropEnumValue:           (*operationGenerator).dropTypeValue,
+	dropSchema:              (*operationGenerator).dropSchema,
+	primaryRegion:           (*operationGenerator).primaryRegion,
+	renameColumn:            (*operationGenerator).renameColumn,
+	renameIndex:             (*operationGenerator).renameIndex,
+	renameSequence:          (*operationGenerator).renameSequence,
+	renameTable:             (*operationGenerator).renameTable,
+	renameView:              (*operationGenerator).renameView,
+	setColumnDefault:        (*operationGenerator).setColumnDefault,
+	setColumnNotNull:        (*operationGenerator).setColumnNotNull,
+	setColumnType:           (*operationGenerator).setColumnType,
+	survive:                 (*operationGenerator).survive,
+	insertRow:               (*operationGenerator).insertRow,
+	selectStmt:              (*operationGenerator).selectStmt,
+	validate:                (*operationGenerator).validate,
+}
+
+var opWeights = []int{
+	addColumn:               1,
+	addConstraint:           0, // TODO(spaskob): unimplemented
+	addForeignKeyConstraint: 0, // Disabled and tracked with #91195
+	addRegion:               1,
+	addUniqueConstraint:     0,
+	alterTableLocality:      1,
+	createIndex:             1,
+	createSequence:          1,
+	createTable:             1,
+	createTableAs:           1,
+	createView:              1,
+	createEnum:              1,
+	createSchema:            1,
+	dropColumn:              0,
+	dropColumnDefault:       1,
+	dropColumnNotNull:       1,
+	dropColumnStored:        1,
+	dropConstraint:          1,
+	dropIndex:               1,
+	dropSequence:            1,
+	dropTable:               1,
+	dropView:                1,
+	dropEnumValue:           1,
+	dropSchema:              1,
+	primaryRegion:           0, // Disabled and tracked with #83831
+	renameColumn:            1,
+	renameIndex:             1,
+	renameSequence:          1,
+	renameTable:             1,
+	renameView:              1,
+	setColumnDefault:        1,
+	setColumnNotNull:        1,
+	setColumnType:           0, // Disabled and tracked with #66662.
+	survive:                 0, // Disabled and tracked with #83831
+	insertRow:               0, // Disabled and tracked with #91863
+	selectStmt:              10,
+	validate:                2, // validate twice more often
+}
+
+// This workload will maintain its own list of supported versions for declarative
+// schema changer, since the cluster we are running against can be downlevel.
+// The declarative schema changer builder does have a supported list, but it's not
+// sufficient for that reason.
+var opDeclarativeVersion = map[opType]clusterversion.Key{
```

Comment:
what's the rationale behind those numbers? For example, why are some `23_1/22_2` while others are `BinaryMinSupportedVersionKey`? 

Also, when we add support for new DDL under declarative schema changer into the workload, we add an entry here and what cluster version should it be (imagine a scenario where we implemented Xxx DDL in 23.2, branch `release-23.2` was cut, and we're trying adding it to this workload in master. What cluster version should we use and is it the same cluster version if we decided to backport this support to `release-23.2` branch?)